OrbitCore/Callstack.h

```diff
@@ -12,15 +12,15 @@
 
 #include "CallstackTypes.h"
 
-struct CallStack {
-  CallStack() = default;
+struct HashedCallstack {
```

Comment:
The "hash" field is just a "cache" for what `Hash()` returns and should probably be a private field as it's an implementation detail. As an implementation detail, it should not affect the name. Why not just `Callstack`?
(Note that in the past we have referred to "hashed callstacks" to mean callstacks that we already had and for which we could use the hash to refer to, so the opposite of this.)

Reply:
We can use a different name if HashedCallstack is not good.
The problem here is that in proto files we have Callstacks which is actually a vector of addresses. Here we refer to Callstack as Callstack + Hash. I guess that using different name (not just `Callstack`) will make it a bit more clear that these types are not equal. But suggestions on better name are always welcome 🙂
And I totally agree that we shouln't use the `hash` field directly, so will make it private

Reply:
Mmh, the feeling I have (this is a point of view I guess) is that the `Hash` method is more of a utility method (it could be a static function taking a callstack as argument) than part of the semantics of the class, and just `Callstack` would be fitting (in other words, the two classes you refer to are indeed basically equal, just that we want to have separate ones, as their scope is quite different and also one is a proto). So this now seems to be a namespace problem, where two different parts of Orbit use different classes with the same name. From this point of view, having the capital S before was a hack but it was convenient. I don't really have a suggestion honestly, other than introducing namespaces or leaving the old name.

Maybe someone else has any thoughts on the matter?

Reply:
I reorganized this struct, could you please take another look? Probably with new changes it makes more sense to leave the previous name (as all data are private now) and add a namespace, but for me it's still a bit weird that in one case Callstack doesn't have `hash` and in another it has.